inference-engine/src/mkldnn_plugin/nodes/mkldnn_priorbox_clustered_node.cpp

```diff
@@ -0,0 +1,170 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "mkldnn_priorbox_clustered_node.h"
+
+#include <algorithm>
+#include <cmath>
+#include <memory>
+#include <vector>
+
+#include <ie_parallel.hpp>
+#include <mkldnn_types.h>
+#include <ngraph/ngraph.hpp>
+#include <ngraph/opsets/opset1.hpp>
+
+using namespace MKLDNNPlugin;
+using namespace InferenceEngine;
+
+bool MKLDNNPriorBoxClusteredNode::isSupportedOperation(const std::shared_ptr<const ngraph::Node>& op, std::string& errorMessage) noexcept {
+    try {
+        const auto priorBox = std::dynamic_pointer_cast<const ngraph::opset1::PriorBoxClustered>(op);
+        if (!priorBox) {
+            errorMessage = "Only opset1 PriorBoxClustered operation is supported";
+            return false;
+        }
+    } catch (...) {
+        return false;
+    }
+    return true;
+}
+
+MKLDNNPriorBoxClusteredNode::MKLDNNPriorBoxClusteredNode(
+    const std::shared_ptr<ngraph::Node>& op,
+    const mkldnn::engine& eng,
+    MKLDNNWeightsSharing::Ptr &cache) : MKLDNNNode(op, eng, cache) {
+    std::string errorMessage;
+    if (!isSupportedOperation(op, errorMessage)) {
+        IE_THROW(NotImplemented) << errorMessage;
+    }
+
+    const auto priorBox = std::dynamic_pointer_cast<const ngraph::opset1::PriorBoxClustered>(op);
+    const ngraph::opset1::PriorBoxClustered::Attributes& attrs = priorBox->get_attrs();
+
+    widths = attrs.widths;
+    heights = attrs.heights;
+    clip = attrs.clip;
+    variances = attrs.variances;
+    step = attrs.step;
+    step_heights = attrs.step_heights;
+    step_widths = attrs.step_widths;
+    offset = attrs.offset;
+
+    number_of_priors = widths.size();
+
+    if (variances.empty()) {
+        variances.push_back(0.1f);
+    }
+}
+
+bool MKLDNNPriorBoxClusteredNode::needShapeInfer() const {
```

Comment:
As you know shape infer means (in `redefineOutputMemory`): 

1. memory description redefinition,
2. posible memory reallocation. 

Can you confirm that we still always need to return `true` here?

Reply:
Agree. That's a valid concern. `redefineOutputMemory` logic is quite heavy. I think we can leave needShapeInfer() impl as is. 
At the same time I expect that `redefineOutputMemory` will not do any memory reallocation in case new output dims are equal with the previous one. In fact this is already true: https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/mkldnn_node.cpp#L550. The only thing that bothers me is that we are creating new desc before that which is quite slow operation: https://github.com/openvinotoolkit/openvino/blob/master/src/plugins/intel_cpu/src/mkldnn_node.cpp#L546. Seems like we can refactor this a little bit in bounds of other PR to reduce the cost for such scenarios. @mandrono what do you think?

Reply:
Will be fixed in this PR: https://github.com/openvinotoolkit/openvino/pull/9338